usr/lib/soft_stdll/soft_specific.c

```diff
@@ -2756,7 +2756,7 @@ static CK_RV softtok_hmac_final(SIGN_VERIFY_CONTEXT *ctx, CK_BYTE *signature,
             if (general)
                 *sig_len = *(CK_ULONG *) ctx->mech.pParameter;
             else
-                *sig_len = (CK_ULONG)mac_len;
+                *sig_len = (CK_ULONG) mac_len;
             return CKR_OK;
```

Comment:
Please move the return CKR_OK outside of this block, so that it also returns when sign is false.

Also, please squash you commits together so that it is only one commit. Currently you have 5 commits on top of each other.